app/backend/wells/views_v2.py

```diff
@@ -573,3 +577,170 @@ class WellDetail(WellDetailV1):
+                When(well_status__well_status_code=WELL_STATUS_CODE_DECOMMISSION, then=F('decommission_end_date')),
+                default=Value(None),
+                output_field=DateField()
+            )
+        )
+
+        return queryset
+
+
+class RecordComplianceListView(ListAPIView):
+    serializer_class = RecordComplianceSerializer
+
+    swagger_schema = None
+    permission_classes = (WellsEditOrReadOnly,)
+    model = Well
+    pagination_class = APILimitOffsetPagination
+
+    # Allow searching on name fields, names of related companies, etc.
+    filter_backends = (WellListFilterBackend, BoundingBoxFilterBackend,
+                       filters.SearchFilter, WellListOrderingFilter, GeometryFilterBackend)
+    ordering = ('well_tag_number',)
+
+    def get_queryset(self):
+        """
+        Retrieves wells that are missing information in any of the specified fields.
+        """
+        queryset = Well.objects.all()
+
+        queryset = Well.objects.select_related('well_status').annotate(
+            work_start_date=Case(
+                When(well_status__well_status_code=WELL_STATUS_CODE_CONSTRUCTION, then=F('construction_start_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_ALTERATION, then=F('alteration_start_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_DECOMMISSION, then=F('decommission_start_date')),
+                default=Value(None),
+                output_field=DateField()
+            ),
+            work_end_date=Case(
+                When(well_status__well_status_code=WELL_STATUS_CODE_CONSTRUCTION, then=F('construction_end_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_ALTERATION, then=F('alteration_end_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_DECOMMISSION, then=F('decommission_end_date')),
+                default=Value(None),
+                output_field=DateField()
+            )
+        )
+
+        # Filtering for records missing any of the specified fields
+        missing_info_filter = (
+            Q(well_tag_number__isnull=True) |
+            Q(identification_plate_number__isnull=True) |
+            Q(well_class__isnull=True) |
+            Q(geom__isnull=True) | # for latitude and longitude
+            Q(finished_well_depth__isnull=True) |
+            Q(surface_seal_depth__isnull=True) |
+            Q(surface_seal_thickness__isnull=True) |
+            Q(aquifer_lithology__isnull=True) |
+            Q(well_status__isnull=True) |
+            Q(work_start_date__isnull=True) |
+            Q(work_end_date__isnull=True) |
+            Q(person_responsible__isnull=True) |
+            Q(company_of_person_responsible__isnull=True) |
+            Q(create_date__isnull=True) |
+            Q(create_user__isnull=True)
+            # Q(natural_resource_region__isnull=True) |
```

Comment:
Should these lines be removed or uncommented?

Reply:
fixed, ty